lesson_9/lesson_9_authentication.py

```diff
@@ -0,0 +1,271 @@
+                    continue
+                break
+            break
+
+    @staticmethod
+    def change_age():
+        check_logins = turn_logins_into_class_user()
+        while True:
+            if not check_logins:
+                print('There are no registered users. You should registrate.')
+                break
+            print('What is your username?')
+            name = ask_name()
+            user = find_user_by_name(name)
+            if not user:
+                print('Error. There is no such user. Try again.')
+                continue
+            while True:
+                print('You need to enter your password in order to change your age.')
+                password = ask_password()
+                if user.password == password:
+                    while True:
+                        new_age = ask_n_check_age()
+                        if new_age == user.age:
+                            print('Error. New age must differ from previous. Try again.')
+                            continue
+                        else:
+                            new_user = User(user.name, user.password, new_age, user.key)
+                            new_user.save(user)
+                            print('Successfully changed your age.')
+                            break
+                else:
+                    print('Incorrect password. Try again')
+                    continue
+                break
+            break
+
+    @staticmethod
+    def change_password():
+        check_logins = turn_logins_into_class_user()
+        while True:
+            if not check_logins:
+                print('There are no registered users. You should registrate.')
+                break
+            print('What is your username?')
+            name = ask_name()
+            user = find_user_by_name(name)
+            if not user:
+                print('Error. There is no such user. Try again.')
+                continue
+            while True:
+                print('You need to enter your key in order to change your password.')
+                key = ask_key()
+                if user.key == key:
+                    while True:
+                        new_password = ask_password()
+                        if new_password == user.password:
+                            print('Error. New password must differ from previous. Try again.')
+                            continue
+                        else:
+                            new_user = User(user.name, new_password, user.age, user.key)
+                            new_user.save(user)
+                            print('Successfully changed your password.')
+                            break
+                else:
+                    print('Incorrect key. Try again')
+                    continue
+                break
+            break
+
+
+class AuthenticationSystem:
+    @classmethod
+    def authentication(cls):
+        check_logins = turn_logins_into_class_user()
```

Comment:
here also looks like duplicating code.. the same alhorithm: `turn_logins..` -> `ask_name` -> `find_user` -> `turn_logins..` and also `find_user` that use `turn_logins..`